vaadin-avatar-flow/src/main/java/com/vaadin/flow/component/avatar/AvatarGroup.java

```diff
@@ -277,6 +277,9 @@ private void deferRegistration(AbstractStreamResource resource) {
         }
 
         private void attachPendingRegistration(Command pendingHandle) {
+            if (getHost().getElement().getNode().isAttached()) {
+                pendingHandle.execute();
```

Comment:
Return here to avoid adding an attach handler that would trigger after detach + attach?

Reply:
I'm not sure if that's okay. The attach listener's registration is used as `pendingRegistration` which is used in a `null` check in `deferRegistration`.

The logic here is surprisingly confusing and I have to admit that I don't fully understand it.

Reply:
Ok, seems that it doesn't matter in this case, and the attach listener is removed after first run (meaning that it's not needed to keep)

Reply:
Done